tinyinst.cpp

```diff
@@ -1125,9 +1125,20 @@ void TinyInst::Init(int argc, char **argv) {
 
   std::list <char *> module_names;
   GetOptionAll("-instrument_module", argc, argv, &module_names);
-  for (auto iter = module_names.begin(); iter != module_names.end(); iter++) {
+#if defined __APPLE__ && defined ARM64
+  std::set <std::string> uniq_mod_names;
+  std::map<std::string, std::vector<std::string>> mod_grp = 
+    parse_dyld_map_file("/System/Library/dyld/dyld_shared_cache_arm64e.map");
+  uniq_mod_names.insert(module_names.begin(), module_names.end());
+  module_names.clear();
+
+  for(const auto& mod_name: uniq_mod_names) {
+    module_names.emplace_back((char*)mod_name.c_str());

```

Comment:
I think it would be a good idea to output a message to user saying which additional modules are instrumented (and why).

Reply:
done